### PR TITLE
feat: [CVS-9140]: Infra service changes

### PR DESCRIPTION
**Pull request by vasuki-kalluri**
##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by vasuki-kalluri** [comment]
retrigger coverage

**Comment by vasuki-kalluri** [comment]
retrigger coverage

**Comment by vasuki-kalluri** [comment]
retrigger coverage

**Comment by vasuki-kalluri** [comment]
retrigger coverage

**Comment by vasuki-kalluri** [comment]
retrigger coverage

**Comment by vasuki-kalluri** [comment]
retrigger coverage

**Comment by vasuki-kalluri** [comment]
retrigger jest

**Comment by vasuki-kalluri** [comment]
retrigger sonar

**Comment by vasuki-kalluri** [comment]
retrigger coverage

**Comment by vasuki-kalluri** [comment]
retrigger coverage

**Comment by vasuki-kalluri** [comment]
retrigger coverage

**Comment by vasuki-kalluri** [comment]
retrigger coverage

**Comment by vasuki-kalluri** [comment]
retrigger coverage

**Comment by vasuki-kalluri** [comment]
retrigger jest

File: src/modules/85-cv/components/HarnessServiceAndEnvironment/components/EnvironmentToolTipDisplay.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  environmentRef,
+  envRefList,
+  color,
+  font,
+  shouldAddEnvPrefix
+}: ToolTipProps): JSX.Element => {
+  const { getString } = useStrings()
+  return (
+    <>
+      {type === 'Infrastructure' ? (
```
Comment (deepesh-ui) [inline src/modules/85-cv/components/HarnessServiceAndEnvironment/components/EnvironmentToolTipDisplay.tsx:25]: NIT :  use Infrastructure from constant

File: src/modules/85-cv/components/HarnessServiceAndEnvironment/components/UseEnvironmentSelectOrCreate/EnvironmentSelectOrCreateHook.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    ],
+    [options]
+  )
+
+  const onSubmit = async (values: any): Promise<void> => {
+    onNewCreated(values)
+  }
+
+  const { openHarnessEnvironmentModal } = useHarnessEnvironmentModal({
+    data: {
```
Comment (deepesh-ui) [inline src/modules/85-cv/components/HarnessServiceAndEnvironment/components/UseEnvironmentSelectOrCreate/EnvironmentSelectOrCreateHook.tsx:36]: This should have be moved to constant

File: src/modules/85-cv/pages/monitored-service/components/Configurations/components/Service/components/MonitoredServiceOverview/MonitoredServiceOverview.tsx [hunk 3]
```diff
@@ -96,31 +102,40 @@ export default function MonitoredServiceOverview(props: MonitoredServiceOverview
...
+                  onNewCreated: newOption => {
+                    if (newOption?.identifier && newOption.name) {
+                      const newEnvOption = { label: newOption.name, value: newOption.identifier }
+                      setEnvironmentOptions([newEnvOption, ...environmentOptions])
+                      updatedMonitoredServiceNameForEnv(formikProps, newEnvOption, formikProps.values?.type)
+                    }
                   }
-                }
-              }}
+                } as EnvironmentMultiSelectOrCreateProps | EnvironmentSelectOrCreateProps
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/monitored-service/components/Configurations/components/Service/components/MonitoredServiceOverview/MonitoredServiceOverview.tsx:124]: Create EnvironmentMultiSelectOrCreateProps | EnvironmentSelectOrCreateProps as a type

**Review by deepesh-ui — APPROVED** [review]

**Review by chiragp1212 — APPROVED** [review]

**Pull request merged**